src/pages/workspaces/workspace/tools/DataStepContent.js

```diff
@@ -0,0 +1,134 @@
+import _ from 'lodash/fp'
+import PropTypes from 'prop-types'
+import { Fragment } from 'react'
+import { div, h, span } from 'react-hyperscript-helpers'
+import { buttonPrimary, RadioButton } from 'src/components/common'
+import DataTable from 'src/components/DataTable'
+import { textInput } from 'src/components/input'
+import Modal from 'src/components/Modal'
+import * as Style from 'src/libs/style'
+import { Component } from 'src/libs/wrapped-components'
+import EntitySelectionType from 'src/pages/workspaces/workspace/tools/EntitySelectionType'
+
+
+export default class DataStepContent extends Component {
+  static propTypes = {
+    entityMetadata: PropTypes.objectOf(PropTypes.shape({
+      count: PropTypes.number.isRequired
+    })).isRequired,
+    entitySelectionModel: PropTypes.shape({
+      newSetName: PropTypes.string.isRequired,
+      selectedElements: PropTypes.array,
+      type: PropTypes.oneOf(_.values(EntitySelectionType))
+    }),
+    onDismiss: PropTypes.func,
+    onSuccess: PropTypes.func,
+    rootEntityType: PropTypes.string,
+    workspaceId: PropTypes.object
+  }
+
+  constructor(props) {
+    super(props)
+    const { rootEntityType, entitySelectionModel } = props
+    this.state = { rootEntityType, entitySelectionModel }
+  }
+
+  setEntitySelectionModel(modelUpdates) {
+    this.setState(({ entitySelectionModel }) => {
+      return { entitySelectionModel: { ...entitySelectionModel, ...modelUpdates } }
+    })
+  }
+
+  isValidSelectionModel() {
+    const { entitySelectionModel } = this.state
+    const { newSetName, selectedEntities, type } = entitySelectionModel
+    const { entityType, name } = selectedEntities
+    return (type === EntitySelectionType.processAll ||
+      (type === EntitySelectionType.processFromSet && !!entityType && !!name) ||
+      (_.size(selectedEntities) > 0 && !!newSetName))
+  }
+
+  render() {
+    const {
+      onDismiss, onSuccess,
+      workspaceId, entityMetadata
+    } = this.props
+    const { entitySelectionModel, entitySelectionModel: { type, selectedEntities, newSetName }, rootEntityType } = this.state
+
+    const count = rootEntityType && entityMetadata[rootEntityType].count
+
+    const isSet = _.endsWith('_set', rootEntityType)
+    const setType = `${rootEntityType}_set`
+    const hasSet = _.has(setType, entityMetadata)
+
+    return h(Modal, {
+      okButton: buttonPrimary({
+        disabled: !this.isValidSelectionModel(),
+        onClick: () => onSuccess(entitySelectionModel)
+      }, 'OK'),
+      onDismiss: onDismiss,
+      width: 'calc(100% - 2rem)'
+    }, [
+      div({ style: { ...Style.elements.sectionHeader, marginBottom: '1rem' } }, [`Select ${rootEntityType}s to process`]),
+      rootEntityType && div({
+        style: {
+          padding: '1rem 0.5rem', lineHeight: '1.5rem'
+        }
+      }, [
+        !isSet && h(Fragment, [
+          div([
+            h(RadioButton, {
+              text: `Process all ${count} rows`,
```

Comment:
nit/TOL: do we want this to change if count == 1 to be grammatically correct?

Reply:
We hacked around this enough in FireCloud that I think we eventually added a (or built a very simple) pluralizer. I think we should do the same here but didn't want to encumber this PR any further. I created issue #1200 for this.